manifests/candlepin.pp

```diff
@@ -91,27 +91,22 @@
       unprotect     => true,
       strip         => true,
       password_file => $ca_key_password_file,
-    } ~>
+    }
+
     certs::keypair { 'tomcat':
-      key_pair  => Cert[$tomcat_cert_name],
-      key_file  => $tomcat_key,
-      cert_file => $tomcat_cert,
-    } ~>
-    file { $keystore_password_path:
-      ensure    => file,
-      content   => $keystore_password,
-      owner     => 'root',
-      group     => $group,
-      mode      => '0440',
-      show_diff => false,
-    } ~>
-    exec { 'candlepin-generate-ssl-keystore':
-      command   => "openssl pkcs12 -export -in ${tomcat_cert} -inkey ${tomcat_key} -out ${keystore} -name tomcat -CAfile ${ca_cert} -caname root -password \"file:${keystore_password_path}\"",
-      unless    => "keytool -list -keystore ${keystore} -storepass:file ${keystore_password_path} -alias tomcat | grep $(openssl x509 -noout -fingerprint -sha256 -in ${tomcat_cert} | cut -d '=' -f 2)",
-      logoutput => 'on_failure',
-      path      => ['/bin/', '/usr/bin'],
-      require   => Keystore[$keystore],
-    } ~>
+      key_pair    => Cert[$tomcat_cert_name],
+      key_file    => $tomcat_key,
+      manage_key  => true,
+      key_owner   => $user,
+      key_group   => $client_keypair_group,
+      key_mode    => '0440',
+      cert_file   => $tomcat_cert,
+      manage_cert => true,
+      cert_owner  => $user,
+      cert_group  => $client_keypair_group,
+      cert_mode   => '0440',
```

Comment:
Did we not manage these before? Did that leave the key world readable?

Reply:
The `privkey` provider ensures a sane default if we do not manage the key through a file resource:

https://github.com/theforeman/puppet-certs/blob/master/lib/puppet/provider/privkey/katello_ssl_tool.rb#L33

```
# ls -la /etc/pki/katello/private/katello-tomcat.key 
-r--------. 1 root root 3243 May 21 13:53 /etc/pki/katello/private/katello-tomcat.key
```